### PR TITLE
dash/unify-tools

### PR DESCRIPTION
**Pull request by karolkolodziej**
Unified tooling between `Dashboards` and `Highcharts`.

To do:

- [x] scripts
- [x] API- dash currently uses the `typedoc` which makes it impossible to unify 
- [x] test
- [x] dist- some unique parts were difficult to unify

### CODE REVIEW

**Comment by TorsteinHonsi** [comment]
Should this PR be marked Draft?

**Review by jomar-honsi — APPROVED** [review]

File: tools/gulptasks/dist-productsjs.js [hunk 1]
```diff
@@ -21,42 +21,50 @@ const TARGET_FILE = Path.join('build', 'dist', 'products.js');
...
-        buildProperties.version ||
-        packageJson.version ||
-        ''
-    ).split('-')[0];
+        products['Highcharts Dashboards'] = {
+            date: [
+                now.getFullYear(),
+                (now.getMonth() + 1).toString().padStart('0', 2),
+                now.getDate().toString().padStart('0', 2)
+            ].join('-'),
```
Comment (sebastianbochan) [inline tools/gulptasks/dist-productsjs.js:52]: `new Date().toISOString().split('T')[0]` seems to be simpler

File: tools/gulptasks/dist-verify.js [hunk 2]
```diff
@@ -6,25 +6,39 @@ const { task } = require('gulp');
...
-        'highmaps/code',
-        'highstock/code'
-    ];
+async function distVerify(argv) {
+    let pathsToVerify;
+
+    if (argv.dashboards) {
+        pathsToVerify = DASHBOARDS_PATHS;
+    } else {
+        pathsToVerify = HIGHCHARTS_PATHS;
```
Comment (sebastianbochan) [inline tools/gulptasks/dist-verify.js:40]: I would simplify it and use `pathsToVerify = argv.dashboards ? DASHBOARDS_PATHS : HIGHCHARTS_PATHS;`

File: tools/gulptasks/lint-dts.js [hunk 3]
```diff
@@ -48,14 +50,19 @@ function task() {
...
-                promiseChain = promiseChain.then(
-                    () => processLib.exec('npx tsc -p ' + folder)
-                );
-            });
+        let directories = fsLib.getDirectoryPaths(TEST_FOLDER, false);
+
+        if (argv.dashboards) {
+            directories = directories.filter(folder => folder.includes('dashboards'));
+        } else {
+            directories = directories.filter(folder => !folder.includes('dashboards'));
```
Comment (sebastianbochan) [inline tools/gulptasks/lint-dts.js:58]: One line condition seems to be more optimized.

**Review by sebastianbochan — CHANGES_REQUESTED** [review]

**Review by sebastianbochan — APPROVED** [review]

**Review by Vazonik — APPROVED** [review]

**Review by TorsteinHonsi — APPROVED** [review]
Awesome, thanks!

**Pull request merged**